src/overlays/actors/ovl_Obj_Timeblock/z_obj_timeblock.c

```diff
@@ -27,41 +57,314 @@ const ActorInit Obj_Timeblock_InitVars = {
+    s32 switchFlag = this->dyna.actor.params & 0x3F;
+    this->unk_172 = globalCtx->msgCtx.unk_E3EC;
+
+    if (this->demoEffectFirstPartTimer > 0 && --this->demoEffectFirstPartTimer == 0) {
+        this->unk_174 = BOOL(Flags_GetSwitch(globalCtx, switchFlag));
+    }
+
+    this->isVisible = ObjTimeblock_CalculateIsVisible(this);
+    this->unk_176 = BOOL(Flags_GetSwitch(globalCtx, switchFlag));
+}
+
+void ObjTimeblock_StartAltBehaviorVisibleActionFunc(ObjTimeblock* this) {
+    this->actionFunc = &ObjTimeblock_AltBehaviorVisibleActionFunc;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA00CC.s")
+void ObjTimeblock_AltBehaviorVisibleActionFunc(ObjTimeblock* this, GlobalContext* globalCtx) {
+    if (this->songObserverFunc(this, globalCtx) && this->demoEffectTimer <= 0) {
+        this->demoEffectFirstPartTimer = 12;
+        ObjTimeblock_SpawnDemoEffect(this, globalCtx);
+        this->demoEffectTimer = 160;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/ObjTimeblock_Init.s")
+        // Possibly points the camera to this actor
+        func_80080480(globalCtx, &this->dyna.actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/ObjTimeblock_Destroy.s")
+        // "◯◯◯◯ Time Block Attention Camera (frame counter  %d)\n"
+        osSyncPrintf("◯◯◯◯ Time Block 注目カメラ (frame counter  %d)\n", globalCtx->state.frames);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA032C.s")
+        ObjTimeblock_ToggleSwitchFlag(globalCtx, this->dyna.actor.params & 0x3F);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA040C.s")
+    func_80BA06AC(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0480.s")
+    if (this->demoEffectTimer == 50) {
+        func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+    }
+    if (!this->isVisible && this->demoEffectTimer <= 0) {
+        ObjTimeblock_StartAltBehaviourNotVisibleActionFunc(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA04F8.s")
+void ObjTimeblock_StartAltBehaviourNotVisibleActionFunc(ObjTimeblock* this) {
+    this->actionFunc = &ObjTimeblock_AltBehaviourNotVisibleActionFunc;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0508.s")
+void ObjTimeblock_AltBehaviourNotVisibleActionFunc(ObjTimeblock* this, GlobalContext* globalCtx) {
+    s32 switchFlag = this->dyna.actor.params & 0x3F;
+    s8 switchFlagIsSet = BOOL(Flags_GetSwitch(globalCtx, switchFlag));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Timeblock/func_80BA0514.s")
+    if (this->unk_176 ^ switchFlagIsSet && switchFlagIsSet ^ BOOL((this->dyna.actor.params >> 15) & 1)) {
+        if (this->demoEffectTimer <= 0) {
+            ObjTimeblock_SpawnDemoEffect(this, globalCtx);
+            this->demoEffectTimer = 160;
```

Comment:
I'm not sure we want to use original names such as "demo" where not necessary (I guess `SpawnDemoEffect` is fine because the actor is still named DEMO_EFFECT), so instead of `demoEffectTimer` have `cutsceneTimer`/`csTimer`?

Actually looking at more of the code I don't think its main usage is about cutscene stuff, looks more like a general purpose timer for delays and "cooldowns" so it would be fine to just call it `timer`.